README.md

```diff
@@ -31,7 +31,140 @@ The known changes include (not comprehensive):
 * adding some sort of env_var type so one does not have to use the `:required` parameter for `#add_flag`
 * the above will allow for multiple commands in a task with independent option, switch, and environment variable tracking
 
-Examples (see the [Use](#use) section for outputs):
+Rototiller has 4 main _types_ of arguments that can be passed to a command in a task. `RototillerTasks` can accept multiple commands.  Each of these argument types has a similar API that looks similar to `add_command()`.
+### All permutations of v2 API (remove and refactor into useful doc sections below upon testing, merge-up to stable)
+
+* all things that can take multiples should use add\_ as precursor to method name
+* all things that only take one should use set\_ as precursor to method name?
+    require 'rototiller'
+
+    ## all task methods
+    rototiller_task :name do |t|
+      t.add_command # t.add_cmd? me no likey
+      t.add_env
+    end
+    rototiller_task do |t|
+      t.set_name = 'string_name' # should this be validated??  e.g.: spaces, etc
+      t.add_command
+      t.add_env
+    end
+
+
+    ## all task's add_env invocations with just name
+    t.add_env('env_name') #required, default messaging
+    t.add_env({:name => 'env_name'})
+    t.add_env :env_name
+    t.add_env 'env_name' # implicitly allowed by ruby
+    t.add_env do |e|
+      e.name
+    end
+
+    ## all task's add_env invocations with name, message
+    #t.add_env('env_name')  # impossible
+    t.add_env({:name => 'env_name', :message})
+    t.add_env :env_name do |e|
+    t.add_env 'env_name' do |e|  # should we do this too?
+      e.set_message
+    end
+    t.add_env do |e|
+      e.name
+      e.message
+    end
+
+    ## all task's add_env invocations with name, value
+    #t.add_env('env_name')  # impossible
+    t.add_env({:name => 'env_name', :default/:value =>}) # optional; default messaging
+    t.add_env :env_name do |e|
+    t.add_env 'env_name' do |e|  # should we do this too?
```

Comment:
no dashes in symbols.  create some conversion inside?  iw, beaker does this with peril